source/fuzz/transformation_composite_extract.cpp

```diff
@@ -133,19 +128,14 @@ void TransformationCompositeExtract::AddDataSynonymFacts(
 
   // Add the fact that the id storing the extracted element is synonymous with
   // the index into the structure.
-  if (!transformation_context->GetFactManager()->IdIsIrrelevant(
-          message_.composite_id())) {
-    std::vector<uint32_t> indices;
-    for (auto an_index : message_.index()) {
-      indices.push_back(an_index);
-    }
-    protobufs::DataDescriptor data_descriptor_for_extracted_element =
-        MakeDataDescriptor(message_.composite_id(), indices);
-    protobufs::DataDescriptor data_descriptor_for_result_id =
-        MakeDataDescriptor(message_.fresh_id(), {});
-    transformation_context->GetFactManager()->AddFactDataSynonym(
-        data_descriptor_for_extracted_element, data_descriptor_for_result_id);
-  }
+  std::vector<uint32_t> indices(message_.index().begin(),
+                                message_.index().end());
+  auto data_descriptor_for_extracted_element =
+      MakeDataDescriptor(message_.composite_id(), indices);
+  auto data_descriptor_for_result_id =
+      MakeDataDescriptor(message_.fresh_id(), {});
+  transformation_context->GetFactManager()->AddFactDataSynonym(
+      data_descriptor_for_extracted_element, data_descriptor_for_result_id);
```

Comment:
It's important to be really careful about irrelevant ids here.  I think the code you're modifying has probably changed since you did this PR; please check carefully that what we're now checking about irrelevant ids is preserved.

Reply:
I removed this condition since it is already checked by the `fuzzerutil::CanMakeSynonymOf` above. Additionally, `AddFactDataSynonym` will have an assertion failure if any of ids descriptors is irrelevant.